charmcraft/commands/store/store.py

```diff
@@ -59,3 +60,15 @@ def whoami(self):
             userid=response['id'],
         )
         return result
+
+    def register_name(self, name):
+        """Register the specified name for the authenticated user."""
+        self._client.post('/v1/charm', {'name': name})
+
+    def list_registered_names(self):
+        """Return names registered by the authenticated user."""
+        response = self._client.get('/v1/charm')
+        result = []
+        for item in response['charms']:
+            result.append(Charm(name=item['name'], private=item['private'], status=item['status']))
```

Comment:
reading the code I have no idea what to expect from `private` and `status`. Are they booleans? integers? strings?

I then go down and look at the tests, and discover that `private` is a boolean, and `status` is a string. Is it an arbitrary string, or is it expected to take on discrete values?

Reply:
This is the [source of truth](https://api.staging.snapcraft.io/publisher/api/docs/default.html) (I now included this in the class' docstring).

`status` is an arbitrary string in the sense it can have multiple values defined by the Store at anytime...

Reply:
Would it be better to make Charm an actual type instead of a namedtuple and then you can more easily docstring/type annotate it?

Reply:
Probably yes, but also to be stricter on the response from the Store (which would be the main benefit). After trying different things and talking about it with @chipaca I ended up opening #70. So, in the future.